### DISCUSSION TITLE
How to disable WebSecurity from WPF?

### DISCUSSION THREAD

**Discussion by Corbie-42** [cefsharp/CefSharp#4568] [Q&A]
I am using CefSharp version 81.3.100.0 (x64, .Net Framework v4.5.2). There is no way I can use a more recent version in my project, so I cannot upgrade.

My goal is to enable **CORS** in the browser, because the websites displayed in CefSharp are calling a server, that does not send the required headers (and I can't change that too).

In a Window, I am creating embedding the browser like this:
```xaml
<cefSharpWpf:ChromiumWebBrowser
        WebBrowser="{Binding Browser, Mode=OneWayToSource}"
        Address="{Binding Address, Mode=OneWay}">
    <cefSharpWpf:ChromiumWebBrowser.BrowserSettings>
        <cefSharpCore:BrowserSettings WebSecurity="Disabled"/>
    </cefSharpWpf:ChromiumWebBrowser.BrowserSettings>
</cefSharpWpf:ChromiumWebBrowser>
```
This initializes the browser as desired, but the `BrowserSettings` in the backing field are still `null`.

**Am I missing something?**

----

I have also tried initializing the `ChromiumWebBrowser` in the backing field, binding the `Browser` above in `Mode=OneWay`:
```csharp
_browser = new ChromiumWebBrowser(Address)
{
    BrowserSettings = new BrowserSettings
    {
        WebSecurity = CefState.Disabled
    }
};
```
But this way the browser never properly initialized (although the website was displayed), so e.g. calling `ShowDevTools` on it failed:
![image](https://github.com/cefsharp/CefSharp/assets/79903698/a08f6f17-cc27-46ed-b8f5-6cdf1845be97)

----

Setting the command line arguments also didn't help:
```csharp
var settings = new CefSettings();
settings.CefCommandLineArgs.Add("disable-web-security", "true");
Cef.Initialize(settings);
```

**Comment by amaitland** [comment]
> My goal is to enable **CORS** in the browser, because the websites displayed in CefSharp are calling a server, that does not send the required headers (and I can't change that too).

Not clear what you are trying to achieve or if disabling web security is actually a solution to your problem.

> This initializes the browser as desired, but the `BrowserSettings` in the backing field are still `null`.

This was fixed, you'd need a newer version to set the settings in `xaml`. You can set the property in code behind.

> But this way the browser never properly initialized

That's not what the error is telling you at all. You need to wait until the browser has been initialized before you start calling methods on the browser. Subscribe to the listed event then call `ShowDevTools`.

> Setting the command line arguments also didn't help:

Load `chrome://version/` in a `ChromiumWebBrowser` instance and check the command line args, if you see `--disable-web-security` then the arg was passed successfully.

> I am using CefSharp version 81.3.100.0 (x64, .Net Framework v4.5.2).

Not supported.
